### PR TITLE
Convert network db to use new async-service API

### PR DESCRIPTION
**Pull request by pipermerriam**
### What was wrong?

The networkdb component isn't using the nice new `async-service` APIs

### How was it fixed?

Converted the component to use the new `async-service` APIs

#### Cute Animal Picture

animal-dogs-in-costumes-s-zone-dogs-christmas-funny-animal-in-costumes-s-zone-pictures-images-photos-pi-pictures-christmas-funny-animal-1024x915

### CODE REVIEW

**Comment by pipermerriam** [comment]
Test failures are only due to a change in logging messages.  I'm going to wait until https://github.com/ethereum/async-service/pull/13 is merged and released to clean this up.

**Comment by pipermerriam** [comment]
Failures need a new release of the `async-service` library to have the proper logging statements available.  Probably something we can get done tomorrow or Wednesday if we get a few of the in-progress bits moved through.

File: tests/integration/test_trinity_cli.py [hunk 1]
```diff
@@ -242,9 +242,9 @@ def amend_command_for_unused_port(command, unused_tcp_port):
         ),
         (   # Enable DEBUG2 logs for everything except discovery which is reduced to ERROR logs
             ('trinity', '-l=DEBUG2', '-l', 'p2p.discovery=ERROR'),
-            {'Started main process', 'ConnectionTrackerServer  Running task <coroutine object'},
+            {'Started main process', '<Manager[ConnectionTrackerServer] SRcfe>: running root task run[daemon=False]'},  # noqa: E501
             {'>>> ping'},
-            {'Started main process', 'ConnectionTrackerServer  Running task <coroutine object'},
+            {'Started main process', '<Manager[ConnectionTrackerServer] SRcfe>: running root task run[daemon=False]'},  # noqa: E501
```
Comment (carver) [inline tests/integration/test_trinity_cli.py]: What's with the `SRcfe`?

Comment (pipermerriam) [reply tests/integration/test_trinity_cli.py]: started-running-cancelled-finished-errored

lowercase = False, uppercase = True

I'm open to other ideas but I previously had it expanded and it resulted in unreadably long logging statements that made it near impossible to figure out what was going on.

Comment (carver) [reply tests/integration/test_trinity_cli.py]: Maybe `flags=SRcfe` could make it a bit more discoverable.

**Pull request merged**